### PR TITLE
Add Decimal arithmetic subtraction function 

### PR DESCRIPTION
**Pull request by karteekmurthys**

### CODE REVIEW

File: velox/functions/prestosql/DecimalArithmetic.cpp [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+namespace {
+
+template <
+    typename R /* Result Type */,
+    typename A /* Argument1 */,
+    typename B /* Argument2 */,
+    typename Operation /* Arithmetic operation */>
+class DecimalBaseFunction : public exec::VectorFunction {
+ public:
+  DecimalBaseFunction(const uint8_t aRescale, const uint8_t bRescale)
```
Comment (majetideepak) [inline velox/functions/prestosql/DecimalArithmetic.cpp:32]: no need of `const` here.

File: velox/functions/prestosql/DecimalArithmetic.cpp [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+        BaseVector::isReusableFlatVector(flat)) {
+      auto flatValues = flat->asUnchecked<FlatVector<R>>();
+      rawResults = flatValues->mutableRawValues();
+      *result = std::move(flat);
+    } else {
+      rawResults = prepareResults(rows, resultType, context, result);
+    }
+    return rawResults;
+  }
+  const uint8_t aRescale_;
```
Comment (majetideepak) [inline velox/functions/prestosql/DecimalArithmetic.cpp:134]: no need for const for scalars.

File: velox/functions/prestosql/DecimalArithmetic.cpp [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+  template <typename R, typename A, typename B>
+  inline static void
+  apply(R& r, const A& a, const B& b, uint8_t aRescale, uint8_t bRescale) {
+    r.setUnscaledValue(
+        a.unscaledValue() * POWERS_OF_TEN[aRescale] +
+        b.unscaledValue() * POWERS_OF_TEN[bRescale]);
+  }
+
+  inline static uint8_t computeRescaleFactor(
+      const uint8_t fromScale,
```
Comment (majetideepak) [inline velox/functions/prestosql/DecimalArithmetic.cpp:149]: these don't have to be const also.

**Review by majetideepak — APPROVED** [review]

File: velox/functions/prestosql/DecimalArithmetic.cpp [hunk 4]
```diff
@@ -125,6 +125,33 @@ class Addition {
...
+      const uint8_t toScale,
+      const uint8_t rScale = 0) {
+    return std::max(0, toScale - fromScale);
+  }
+
+  inline static const std::pair<uint8_t, uint8_t> computeResultPrecisionScale(
```
Comment (Yuhta) [inline velox/functions/prestosql/DecimalArithmetic.cpp:145]: Returned rvalue does not need `const`

File: velox/functions/prestosql/DecimalArithmetic.cpp [hunk 5]
```diff
@@ -201,4 +228,8 @@ VELOX_DECLARE_STATEFUL_VECTOR_FUNCTION(
     udf_decimal_add,
     decimalAddSubtractSignature(),
     createDecimalFunction<Addition>);
+VELOX_DECLARE_STATEFUL_VECTOR_FUNCTION(
```
Comment (Yuhta) [inline velox/functions/prestosql/DecimalArithmetic.cpp:231]: nit: Would be nice to have an empty line above

File: velox/functions/prestosql/DecimalArithmetic.cpp [hunk 6]
```diff
@@ -135,14 +135,12 @@ class Subtraction {
         b.unscaledValue() * DecimalUtil::kPowersOfTen[bRescale]);
   }
 
-  inline static uint8_t computeRescaleFactor(
-      const uint8_t fromScale,
```
Comment (majetideepak) [inline velox/functions/prestosql/DecimalArithmetic.cpp]: removed the `const` from arguments by mistake?

Comment (majetideepak) [reply velox/functions/prestosql/DecimalArithmetic.cpp]: I see this is in line with the Addition.

**Comment by majetideepak** [comment]
@Yuhta can you import again? Thanks.

**Pull request closed without merging**